pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -2430,3 +2434,30 @@ func (a *ApplicationBackupController) createVMIncludeResources(backup *stork_api
 func IsBackupObjectTypeVirtualMachine(backup *stork_api.ApplicationBackup) bool {
 	return backup.Spec.BackupObjectType == resourcecollector.PxBackupObjectType_virtualMachine
 }
+
+func (a *ApplicationBackupController) pruneNamespacesForVmBackup(backup *stork_api.ApplicationBackup) bool {
+	nsList := make([]string, 0)
+	if len(backup.Spec.Namespaces) == 0 {
+		return false
+	}
+	if len(a.vmIncludeResource[string(backup.UID)]) != 0 {
+
+		nsMap := make(map[string]bool)
+		for _, obj := range a.vmIncludeResource[string(backup.UID)] {
```

Comment:
My understanding is `vmIncludeResource    map[string][]stork_api.ObjectInfo` this already has the vmlist from the needed ns, can we iterate over `[]stork_api.ObjectInfo` and get the ns list.

Reply:
Logic changed, resolving this comment.